bash_completion

```diff
@@ -552,10 +552,10 @@ _quote_readline_by_ref()
         # to drop the additional quoting.  See also:
         # https://www.mail-archive.com/bash-completion-devel@lists.alioth.debian.org/msg01942.html
         if [[ ${!2} == \$\'*\' ]]; then
-          local value=${!2:2:-1}   # Strip beginning $' and ending '.
-          value=${value//'%'/%%}   # Escape % for printf format.
-          printf -v value "$value" # Decode escape sequences of \....
-          local "$2" && _upvars -v "$2" "$value"
+            local value=${!2:2:-1}      # Strip beginning $' and ending '.
+            value=${value//'%'/%%}      # Escape % for printf format.
+            printf -v value %s "$value" # Decode escape sequences of \....
```

Comment:
This change to `printf` breaks the behavior. Here, `printf -v value "$value"` is intentionally used to process escape sequences of the form `\...` inside the variable `value`. Actually, if we would write it as `printf -v value %s "$value"`, it would be just equivalent to `value=$value` so would do nothing. `shellcheck` reports it because typical wrong codes try to use variables in the format string in order to output their values *literally*, but this is not the case this time. Maybe we can add `# shellcheck disable=SC2059`

Reply:
Ok. It's important to run the linters and add required disables to make such nuances explicit. Also, this doesn't seem to be covered in the test suite (no errors after my change), and I'm unable to quickly create a case that this would break. If you have one at hand, could you submit a PR with the test change (and the fix), or let me know how to provoke it otherwise?

Reply:
OK. I'll try to make a PR. Thank you

Reply:
cf. PR #526